Library/OcCpuLib/OcCpuLib.c

```diff
@@ -884,27 +880,33 @@ ScanAmdProcessor (
         //
         if (Cpu->MaxExtId >= 0x8000001E) {
           AsmCpuid (0x8000001E, NULL, &CpuidEbx, NULL, NULL);
-          Cpu->CoreCount =
-            (UINT16) DivU64x32 (
-              Cpu->ThreadCount,
-              (BitFieldRead32 (CpuidEbx, 8, 15) + 1)
-            );
+          Cpu->CoreCount = (UINT16)DivU64x32 (Cpu->ThreadCount, (BitFieldRead32(CpuidEbx, 8, 15) + 1));
         }
         break;
-      case 0x06:
+      case 0x06: // 15h_16h Family
       case 0x07:
         CofVid           = AsmReadMsr64 (K10_COFVID_STATUS);
         CoreFrequencyID  = BitFieldRead64 (CofVid, 0, 5);
         CoreDivisorID    = CofVid & BIT6;
         Divisor          = LShiftU64 (1, CoreDivisorID);
-        Cpu->MaxBusRatio = (UINT8)(((CoreFrequencyID + 16) / Divisor) + 1);
+        Cpu->MaxBusRatio = (UINT8) ((CoreFrequencyID + 16) / (2 * Divisor));
         //
         // AMD 15h and 16h CPUs don't support hyperthreading,
         // so the core count is equal to the thread count
+        //
         Cpu->CoreCount = Cpu->ThreadCount;
         break;
       default:
         return;
+	
+	  DEBUG ((
+	  DEBUG_INFO,
```

Comment:
This change makes no sense to me, as the code is never executed due to `return` above. Most likely it should be outside of `switch`. Formatting arguments are all wrong: UINT64 is printed with `Lu` not `d`. In addition to that you for some reason used tabulation symbols to indent the lines, please follow EDK II codestyle and use spaces only.

Reply:
thanks for spotting another mistake i will correct it and will also follow EDK II code style.

Reply:
oh! sorry .
yea its need to be outside of switch .
Thanks will correct it : )